### ISSUE TITLE
SASRec baseline

### ISSUE THREAD

**Issue by Janetalready** [QYQ-bot/CLEA#3]
Hi! I got two questions regarding the SASRec baseline.

1. for each user, do you combine items from all the baskets into a sequence and the input is a sequence of **items** or you do the average pooling over each basket and the input of the SASRec is a sequence of **sets**

2. If the input is a sequence of items, "For SASRec, we set the maximum length of a basket sequence to 20 (i.e., 𝑛 = 20)", are you saying the length of the model input equals 20(20 items)?

**Comment by GodofRap** [comment]
For your 1st point, according to the paper, I believe it is transferred into a sequence of items and the items in the same basket having the same positional embedding.

**Comment by Janetalready** [comment]
Thanks! That makes sense. Since I still cannot reproduce the results for SASRec, may I also know your hyper parameter settings for SASRec(max sequence length, learning rate, weight decay, number of attention head, number of layer, hidden size, drop out rate, training epochs)? By the way, did you trunk each basket to a limit size for SASRec, like there are at most 35 items for each basket?

**Comment by Janetalready** [comment]
Hi, another follow-up question, did you also use 1000 negative items for SASRec?

**Comment by GodofRap** [comment]
> Thanks! That makes sense. Since I still cannot reproduce the results for SASRec, may I also know your hyper parameter settings for SASRec(max sequence length, learning rate, weight decay, number of attention head, number of layer, hidden size, drop out rate, training epochs)? By the way, did you trunk each basket to a limit size for SASRec, like there are at most 35 items for each basket?

Hi Jan, have you reproced the results for SASRec? I'm also working on this and still encounter some problem.

**Comment by Janetalready** [comment]
> > Thanks! That makes sense. Since I still cannot reproduce the results for SASRec, may I also know your hyper parameter settings for SASRec(max sequence length, learning rate, weight decay, number of attention head, number of layer, hidden size, drop out rate, training epochs)? By the way, did you trunk each basket to a limit size for SASRec, like there are at most 35 items for each basket?
> 
> Hi Jan, have you reproced the results for SASRec? I'm also working on this and still encounter some problem.

Hi @GodofRap, I followed the procedure but can get at most 24% for now. There could still be some details misaligned....

**Comment by GodofRap** [comment]
The same here. I cannot achieve such high performance either. I believe there are some magic not contained in the papers.

**Comment by GodofRap** [comment]
By the way, do you mean your result is on the Instcart data?

**Comment by Janetalready** [comment]
> By the way, do you mean your result is on the Instcart data?

Yes

**Comment by Janetalready** [comment]
I'm still not sure about my data processing part. For example, I set the maximum sequence length to 5. Given a basket sequence [[1,2],[3]] and the next basket [4,5], there will be two item sequences for training 1) Input: [-1, -1, 1, 2, 3]. Target sequence is one time-step forward: [-1, -1, 2, 3, 4]. Negative sampled sequence: [-1, -1, 10, 9, 8];  2) Input:[-1, -1, 1, 2, 3]. Target sequence: [-1, -1, 2, 3, 5]. Negative sampled sequence: [-1, -1, 12, 10, 7].  Is this the right way of constructing training sequences? Thanks!

**Comment by Janetalready** [comment]
And the positional embedding for both is [0, 0, 0, 0, 1]

**Comment by Janetalready** [comment]
By the way,  I download the original Instacart dataset and construct the basket sequence resembling your user_date_tran_dict_new.txt file. But I cannot get the similar data statistics reported in table 1. Is it possible for you to provide the data processing code?

**Comment by Janetalready** [comment]
That makes sense. Thanks!

**Comment by Janetalready** [comment]
Is it possible for you to also provide the data processing scripts of other datasets?

**Comment by GodofRap** [comment]
> That makes sense. Thanks!
Have you been able to reproduce the results now?
